### DISCUSSION TITLE
Name of the backup file

### DISCUSSION THREAD

**Discussion by blwhite7** [Authenticator-Extension/Authenticator#758] [Q&A] [answered]
Unfortunately, after a computer issue, I was forced to start afresh with a new macOS. This wiped all of my existing codes. I had made a backup in the past, but I can't find the file. What are the possible default file names for the backup?

**Comment by Sneezry** [comment, marked as answer]
It's `authenticator.json` (Encrypted) or `authenticator.txt` (Unencrypted).

**Reply by blwhite7** [reply]
As a follow-up to this, where is the service actively drawing credentials from? Is there a file path I could look at in a laptop backup (from when the credentials were known to be working properly) and replace this file on my current machine?

**Reply by blwhite7** [reply]
I am on mac for context

**Reply by mymindstorm** [reply]
It is stored as part of your browser profile data, but in a file format that is hard to extract data from or replace on another computer.

**Reply by blwhite7** [reply]
@mymindstorm do you know what the relevant file(s) that contain the information are? I named my local authenticator backup something other than the default and can't find it... might be worth a shot to use time machine to replace the relevant files

**Reply by mymindstorm** [reply]
You'd need to find the folder with your chrome/Firefox profile and migrate it to the new device. That isn't supported nor something that I can help you with, especially since I don't know where it is stored on a Mac. I think it would be much less trouble to restore another way.

This was all I was able to find: https://superuser.com/questions/1660250/google-chrome-macos-transfer-all-user-profiles-and-passwords

**Reply by blwhite7** [reply]
This might actually be a bug... I have poked around for a couple of hours, and I am finding in `Chrome > Default > Local Extension Settings` and `Chrome > Default > Sync Extension Settings` that there is a file, `000003.log` present in both directories that contains all of the information necessary to manually create the backup file you suggest. 

In my case, `000003.log` in `Sync Extension Settings` does not match `000003.log` in `Local Extension Settings` and only the credential from `Sync Extension Settings` is being displayed by the chrome extension. `000003.log` in `Local Extension Settings` contains all of the passwords I was anticipating upon restoring the computer.

Is this expected? Would it not be desirable to reconcile the local and synced credentials?

**Reply by mymindstorm** [reply]
If you open a debug console for the extension, execute `localStorage.storageLocation = "local"` and it show the correct data.

**Comment by blwhite7** [comment]
Thank you!
